tests/Doctrine/Tests/DBAL/Functional/ConnectionTest.php

```diff
@@ -366,6 +367,21 @@ public function testUserProvidedPDOConnection(): void
         self::assertInstanceOf(ForwardCompatibility\Result::class, $result);
     }
 
+    /**
+     * @requires extension pdo_sqlite
+     */
+    public function testUserProvidedPDOConnectionWithoutDriverManager(): void
+    {
+        $connection = new Connection(
+            ['pdo' => new PDO('sqlite::memory:')],
+            new PDODriver()
+        );
+
+        $result = $connection->executeQuery('SELECT 1');
+
+        self::assertInstanceOf(ForwardCompatibility\Result::class, $result);
```

Comment:
I believe the expected type should be one of these two: https://github.com/doctrine/dbal/blob/63d19ff4f3ac3eafe357e11b4fa11f13020b75a6/lib/Doctrine/DBAL/Connection.php#L1279

Otherwise, if the code starts returning just the `Result`, the method signature will be violated but the test will pass.